### ISSUE TITLE
`crayon.enabled` is never set back

### ISSUE THREAD

**Issue by dipterix** [ropensci/targets#1210]
## Prework

* [X] I understand and agree to [help guide](https://books.ropensci.org/targets/help.html).
* [X] I understand and agree to [contributing guide](https://github.com/ropensci/targets/blob/main/CONTRIBUTING.md).
* [X] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "trouble" or "other" issue so we can discuss your use case and search for existing solutions first.

## Proposal

Current [`callr_dispatch`](https://github.com/ropensci/targets/blob/8ab431908ea584f990fc2727cd4e2aaabe3d0cb2/R/utils_callr.R#L52-L88) changes `crayon.enabled` option which may be used by other programs.

Could you set it back once the function exits?
 
```r
...
oldv = getOptions("crayon.enabled", interactive())
on.exit({
  options(crayon.enabled = oldv)
})
```

**Comment by wlandau** [comment]
Thanks, should be fixed in 5aa33e91cfeca3d4dd8f2f05c36d9f3164863d27. 

Options are already set and reset in `tar_callr_inner_try()` and should only be set inside the `callr` process. (But `crayon.enabled` is an old deprecated option at this point, is I removed it anyway.)

Turns out the culprit was in `tar_error()` and `tar_warning()`, where `on.exit(opionts(old))` was missing an `add = TRUE`.

**Comment by dipterix** [comment]
Sounds great, thanks.
